test/e2e.method.call.js

```diff
@@ -62,8 +62,14 @@ describe('method.call [ @E2E ]', function () {
                 assert.fail();
 
             } catch (err) {
-                assert(err.message.includes("Returned values aren't valid"));
-                assert(err.message.includes('the correct ABI'));
+                // ganache | geth <= 1.9.13
+                const nullDataResponse = err.message.includes("Returned values aren't valid") &&
+                                         err.message.includes('the correct ABI');
+
+                // geth >= 1.9.15
+                const gethErrResponse = err.message.includes("revert");
+
+                assert(nullDataResponse || gethErrResponse);
```

Comment:
There's another behavioral change in 1.9.15. Geth now responds to this ABI mismatch case with:
```
{ 
  jsonrpc: '2.0',
  id: 8,
  error: { code: -32000, message: 'execution reverted' } 
}
```

Previously it returned 
```
{ 
  jsonrpc: '2.0', 
  id: 8, 
  result: '0x' 
}
```
which Web3 interprets as an invalid response, erroring with the message.
```
Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using 
the correct ABI for the contract you are retrieving data from, requesting data from a block number 
that does not exist, or querying a node which is not fully synced
``` 

We *can* detect this case and preserve the existing behavior but am unsure what to do here. 

**Questions**
+ Should we pass client errors as they are and let the clients manage their own error UX? 
+ ..or should we try to correct what seems like substantial reduction in the error message's clarity/usefulness?

Reply:
I like the idea of preserving the existing error, although maybe it could be split up into more specific errors if possible, or at least reworded with better grammar.

I also like the idea of passing on client errors when we can, as I imagine it could be helpful in debugging.

Reply:
One concern I have is that the new geth message doesn't seem right in this context and they might change it. The additional handling on our side basically involves grepping their message and routing the error handling back through the old path.

In this case, we've sent a message that can't be executed...not positive but I don't think this is "reverting" per se. Will ask for further guidance over there.

Reply:
For the mismatched ABI case, clients are responding as follows:

### "0x" data / non-error responses
**geth <= 1.9.14**
```
{
 "jsonrpc": "2.0",
 "id": 1002,
 "result": "0x"
}
```

**ganache-cli --noVMErrorsOnRPCResponse** (geth emulator)
```
{
 "id": 1002,
 "jsonrpc": "2.0",
 "result": "0x"
}
```

### Error responses
**geth >= 1.9.15**
```
{
 "jsonrpc": "2.0",
 "id": 1002,
 "error": {
  "code": -32000,
  "message": "execution reverted"
 }
}
```

**buidler node**
```
{
 "jsonrpc": "2.0",
 "id": 1002,
 "error": {
  "code": -32003,
  "message": "Transaction reverted without a reason"
 }
}
```

**ganache-cli (default)**
```
{
 "id": 1002,
 "jsonrpc": "2.0",
 "error": {
  "message": "VM Exception while processing transaction: revert",
  "code": -32000,
  "data": {
   "0xe60e50d8bb9a9190c651d6ae30bc44bedc217c10ac83ef3251a4d121f09a8d54": {
    "error": "revert",
    "program_counter": 77,
    "return": "0x"
   },
   "stack": "o: VM Exception while processing transaction: revert\n    at Function ..."
  }
}
```

**Parity/OpenEthereum** (via kovan)
```
{
 "jsonrpc": "2.0",
 "id": 1002,
 "error": {
  "code": -32015,
  "data": "Reverted 0x",
  "message": "VM execution error."
 }
}
```

So, I think I was wrong about the new message being incorrect since the majority of responses report this as a `revert`. And the formats have a lot of variety. 

Am inclined towards forwarding errors as they are after seeing this. It's not clear how we'd distinguish wrong ABI from an intentional revert invocation in the code.